src/ResourceManager/Resources/Commands.Resources/RoleAssignments/ValidateScope.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.Common
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public class ValidateScope : ValidateArgumentsAttribute
+    {
+        protected override void Validate(object arguments, EngineIntrinsics engineIntrinsics)
+        {
+            if (arguments != null)
+            {
+                string scope = (string)arguments;
+                var pairs = scope.Split('/');
+                if (pairs.Count() % 2 != 0)
```

Comment:
>pairs [](start = 20, length = 5)

I wonder if I should put more strict check here, e.g. check if the name at odd index falls into some constant strings ( like: "subscriptions")